### PR TITLE
docs(app): light-shadows and shaders docs examples

### LINKED ISSUE #476: [Docs] - How to enable shadows and lights.
### Description

I think it would be insightful for new joiners to have a page explaining how to enable shadows and set which components cast and receive shadows.

> 

### Suggested solution

Add a page under `Examples/`

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [X] I agree to follow this project's [Code of Conduct](https://github.com/Tresjs/tres/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/Tresjs/tres/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://tresjs.org/guide).
- [X] Check that there isn't [already an issue](https://github.com/tresjs/tres/issues) that reports the same bug to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by JaimeTorrealba**
closes #476

### CODE REVIEW

File: docs/examples/shaders.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+And that it is, we have our basic shader running smoothly.
+
+## Using GLSL vite-pluging (optional)
+
+_This step is completly optional and is out of the scope of the **TresJs** team_
+
+Defining our shader inline is not always the best idea, but if you're using VITE you can put your `GLSL` files in a different file just by using the (vite-plugin-glsl)[https://www.npmjs.com/package/vite-plugin-glsl] (check out the link for the official documentation).
+
+And you could have a structure similar to this:
+
```
Comment (alvarosabu) [inline docs/examples/shaders.md:165]: You might want to wrap this on code ` somehow, is not rendering as you are expecting 

Screenshot 2024-01-12 at 16 41 49

File: docs/examples/shaders.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+As you know every instance in [ThreeJs](https://threejs.org/) is available in **TresJs**, so is the `ShaderMaterial`, we just need to add the `Tres` prefix to use it.
+
+For our blob, we could use a simple `SphereGeometry` adding some widthSegments and heightSegments to create a smooth effect, and put our blob 4 units in the Y positive axis
+
+```vue
+    <TresMesh :position="[0, 4, 0]">
+      <TresSphereGeometry :args="[2, 32, 32]" />
+      <TresShaderMaterial />
+    </TresMesh>
+```
```
Comment (alvarosabu) [inline docs/examples/shaders.md:53]:
```suggestion
```vue
<TresMesh :position="[0, 4, 0]">
  <TresSphereGeometry :args="[2, 32, 32]" />
  <TresShaderMaterial />
</TresMesh>
```
```

File: docs/examples/shaders.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
+# Shaders
+
+This guide will help you get started with shaders in TresJS.
+
+We will build a simple scene with a blob. We will then animate the blob to softly distorted it.
+
+_Basic knowledge of how shaders work is necessary_
```
Comment (alvarosabu) [inline docs/examples/shaders.md:7]:
```suggestion
::: warning
_Basic knowledge of how shaders work is necessary
:::
```

File: docs/.vitepress/config.ts [hunk 4]
```diff
@@ -73,6 +73,8 @@ export default defineConfig({
           { text: 'Load Textures', link: '/examples/load-textures' },
           { text: 'Load Models', link: '/examples/load-models' },
           { text: 'Load Text', link: '/examples/text-3d' },
+          { text: 'Lights Shadows', link: '/examples/lights-shadows' },
```
Comment (alvarosabu) [inline docs/.vitepress/config.ts:76]:
```suggestion
          { text: 'Lights & Shadows', link: '/examples/lights-shadows' },
```

File: docs/.vitepress/config.ts [hunk 5]
```diff
@@ -73,6 +73,8 @@ export default defineConfig({
...
+          { text: 'Using Shaders', link: '/examples/shaders' },
```
Comment (alvarosabu) [inline docs/.vitepress/config.ts:77]:
```suggestion
          { text: 'Shaders', link: '/examples/shaders' },
```

**Review by alvarosabu — CHANGES_REQUESTED** [review]

**Review by alvarosabu — APPROVED** [review]

File: docs/examples/lights-shadows.md [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+  <TresDirectionalLight
+    cast-shadow
+    :position="[0, 2, 0]"
+    :intensity="1"
+  />
+  
+  //...
+</template>
+```
+**Number 3**: Set the objects to cast or receive shadows
```
Comment (alvarosabu) [inline docs/examples/lights-shadows.md:111]:
```suggestion
###  Set the objects to cast or receive shadows
```

**Comment by JaimeTorrealba** [comment]
@alvarosabu thanks for pay attention to these details, I believe we're ready, let me know

File: docs/examples/lights-shadows.md [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+
+Internally, ThreeJS automatically generates a new mesh with a [ShadowMaterial](https://threejs.org/docs/index.html?q=shado#api/en/materials/ShadowMaterial) which gets updated in each frame, that is why if you apply animations, the shadow also is animated, but also why you have to use shadows carefully, because they could slow your performance down.
+
+::: warning
+The overuse of shadows in this way could drop your performance. However, there are ways to increase your performance, for more information please check out [this video](https://youtu.be/WGNvVGrS0kY?si=q7XyL5eABKUh3gbS&t=1256)
+:::
+
+## Setting our lights and shadow
+
+We could divide this into three steps.
```
Comment (alvarosabu) [inline docs/examples/lights-shadows.md:79]:
```suggestion
We could divide this into three steps:
```

File: docs/examples/lights-shadows.md [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+
+### Set the shadows on the renderer
```
Comment (alvarosabu) [inline docs/examples/lights-shadows.md:81]:
```suggestion
### Activate shadows on the renderer
```

File: docs/examples/lights-shadows.md [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+
+There are also many types of shadows, for example the "soft shadow" is generated automatially when an object receives more light from one side, but in summary a "ThreeJS default shadow" that is directed towards another surface needs to be cast by a mesh and another mesh needs to receive it. As we see in our example, the `Plane` is receiving a shadow but not casting it. Please note that not all materials can cast or receive shadows.
+
+Internally, ThreeJS automatically generates a new mesh with a [ShadowMaterial](https://threejs.org/docs/index.html?q=shado#api/en/materials/ShadowMaterial) which gets updated in each frame, that is why if you apply animations, the shadow also is animated, but also why you have to use shadows carefully, because they could slow your performance down.
+
+::: warning
+The overuse of shadows in this way could drop your performance. However, there are ways to increase your performance, for more information please check out [this video](https://youtu.be/WGNvVGrS0kY?si=q7XyL5eABKUh3gbS&t=1256)
+:::
+
+## Setting our lights and shadow
```
Comment (alvarosabu) [inline docs/examples/lights-shadows.md:77]:
```suggestion
## Enabling shadows
```

File: docs/examples/lights-shadows.md [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+<template>
+  <TresCanvas
+    clear-color="#111"
+    shadows
+    window-size
+  />
+  //...
+</template>
+```
+### Set the light to cast shadow
```
Comment (alvarosabu) [inline docs/examples/lights-shadows.md:95]:
```suggestion
### Set the light to cast shadows
```

**Review by alvarosabu — CHANGES_REQUESTED** [review]
@JaimeTorrealba way better now, I just added some quick suggestions for the wording of the Shadows setup, we were using the same terms everywhere, was redundant to read.

**Pull request merged**